public/js/graph/react_graph.js

```diff
@@ -554,6 +556,96 @@ var Graph = React.createClass({
         }
     },
 
+    /**
+     * Calculates the position of the click in relation to the page.
+     * @param {object} e The click event.
+     * @param {HTMLElement} elem The target of the click event.
+     * @return {object} The position of the click.
+     */
+    getClickPosition: function(e, elem) {
```

Comment:
Because these functions don't need `this`, you can make them helpers which are outside of the class. (Put them at the bottom of the file for now.)